modules/ROOT/pages/access-source-message-attributes.adoc

```diff
@@ -0,0 +1,146 @@
+= Accessing Source Message Attributes in Translation Maps
+
+*Vijay, I am finding the text describing the use cases awkward when I use the term "accessing." To me, you are adding source attributes to translation maps or maybe exposing them, but not really accessing them. If you agree with "Exposing Source Message Attributes in Translation Maps,"" I will change the use case wording accordingly.* 
+
+When you create translation maps for inbound or outbound message flows, you can access source message attributes in those maps. The source attributes in the maps appear in the translated message payload, and you can view them on the activity screen for transmissions that use the map. 
+
+Some common use cases include:
+
+* Achieving end-to-end tracing
++
+Accessing transmission IDs in the translation maps enables you to achieve end-to-end tracing becuase these IDs are passed to the translated JSON or XML output file.
++
+* Mapping attributes to pass them to the backend
++
+For example, if the backend must see the source filename or a specific HTTP header, you can access the filename or HTTP header in the translation map so that it appears in the translated data payload.
++
+* Applying mapping logic based either on a filename part or pattern within the transformation
++
+For example, you can access a filename prefix through substring operations and can build dynamic mapping logic based on this prefix. 
++
+* Mapping attributes to make them available in the search
++
+For example, you can access the source filename or a specific HTTP header in the translation map so that you can easily search for these values in the Activity screens.
+
+The source attributes that you can access are available within a predefined variable named `b2bMessageContext`. The content of this variable varies based on the source endpoint's protocol.
+
+== AS2 Source Endpoint
+
+Messages received via AS2 contain the following fields in the `b2bMessageContext` variable:
+
+* sourceProtocol
+* transmissionId
+* sourceEndpointAttributes:
+** as2MessageId
+** as2From
+** as2To
+** fileName
+** subject
+** fileSize 
+* customProtocolHeaders
++
+These headers always begin with the prefix `x-apm-`.
```

Comment:
Please change as follows:
For security reasons, only the headers with the prefix 'x-apm-' are made available in the 'b2bMessageContext' variable.